### DISCUSSION TITLE
How to utilize NuxtLink when parsing links in markdown files

### DISCUSSION THREAD

**Discussion by manastalukdar** [nuxt/nuxt#26574] [Questions]
I am loading markdown files into my nuxt3 based website using markdown-it and some custom rendering. When links in markdown are pointing to locations within the website itself, I would like in some way to utilize the NuxtLink capability. I tried to write a custom renderer extension on top of markdown-it that can possible replace the anchor element but not even sure if that will work. Any pointers are appreciated.

**Comment by tubstrr** [comment]
Check out Vue's documentation on [Render Functions & JSX](https://vuejs.org/guide/extras/render-function.html#render-functions-jsx)

I ran into a similar issue with a headless Nuxt build I was working on, my CMS's WYSIWYG editor would hard code all `<a>` tags. In my case I wrote a custom HTML component, that would replace all `<a>` tags with `<NuxtLink>`

These 2 packages did most of the heavy lifting for SSR:
- [html-entities](https://www.npmjs.com/package/html-entities)
- [node-html-parser](https://www.npmjs.com/package/node-html-parser)
